scripts/perftune.py

```diff
@@ -759,12 +802,24 @@ def __setup_one_hw_iface(self, iface):
         #
         # If this NIC has a limited number of Rx queues then we want to distribute their IRQs separately.
         # For such NICs we've sorted IRQs list so that IRQs that handle Rx are all at the head of the list.
+        perftune_print("Number of Rx queues: {}".format(max_num_rx_queues))
         if max_num_rx_queues < len(all_irqs):
             num_rx_queues = self.__get_rx_queue_count(iface)
             perftune_print("Distributing IRQs handling Rx:")
             distribute_irqs(all_irqs[0:num_rx_queues], self.irqs_cpu_mask)
             perftune_print("Distributing the rest of IRQs")
             distribute_irqs(all_irqs[num_rx_queues:], self.irqs_cpu_mask)
+
+            #use ethtool to tune down rx queues if user wants to do it
+            if self.args.tune_rxqueues:
+                queues_count_from_cpuset = bin(
+                    int(self.irqs_cpu_mask, 16)).count("1")
+                qmax, qcurrent = self.__get_current_rx_queues(iface)
+                new_rx_queues_count = queues_count_from_cpuset if qmax > queues_count_from_cpuset else qmax
+                if qcurrent != new_rx_queues_count:
+                    self.__set_driver_rx_queues(iface, new_rx_queues_count)
+                perftune_print("Distributing IRQs handling Rx after limiting the rx hw queues {} :".format(new_rx_queues_count))
+                distribute_irqs(all_irqs[0:new_rx_queues_count], self.irqs_cpu_mask)
```

Comment:
You are at the wrong clause. The condition at the line 806 is detecting NICs that have a limitation on the max RSS count which is less than the total number of IRQs present.

The trimming Rx queues down logic covers all NICs and hence should be outside and before this clause.

In particular we should trim Rx queues, make sure that `max_num_rx_queues` reflects that and only then continue to `distribute_irqs` calls.